File: Data Structures & Algorithms/task-scheduling/submission-1.cpp
class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
        std::vector<int> counts (26, 0);
        int max_frequency = 0;
        for (const char& task : tasks) {
            counts[task - 'A']++;
            if (max_frequency < counts[task - 'A']) {
                max_frequency = counts[task - 'A'];
            }
        }

        int max_frequency_count = 0;
        for (const int& count : counts) {
            if (count == max_frequency) max_frequency_count++;
        }

        return std::max((int)tasks.size(), (max_frequency - 1) * (n + 1) + max_frequency_count);
    }
};

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cs
public class Solution {
    public int CharacterReplacement(string s, int k) {
        // window size - mostFrequence <= k
        int[] count = new int[26];
        int mostFrequence = 0;
        int l = 0;
        int length = 0;

        for (int r = 0; r < s.Length; r++) {
            count[s[r] - 'A']++;
            mostFrequence = Math.Max(mostFrequence, count[s[r] - 'A']);

            while ((r - l + 1) - mostFrequence > k) {
                count[s[l] - 'A']--;
                l++;
            }

            length = Math.Max(length, r - l + 1);
        }

        return length;
    }
}
